packages/manager/modules/pci/src/projects/project/storages/cold-archive/users/add/user-add/user-add.controller.js

```diff
@@ -0,0 +1,173 @@
+import {
+  NAMESPACES,
+  TRACKING,
+  USER_STATUS,
+  OBJECT_STORAGE_USER_ROLE,
+} from '../../users.constants';
+
+export default class PciUsersAddController {
+  /* @ngInject */
+  constructor(
+    $state,
+    $translate,
+    atInternet,
+    CucCloudMessage,
+    PciStoragesColdArchiveService,
+  ) {
+    this.$state = $state;
+    this.$translate = $translate;
+    this.atInternet = atInternet;
+    this.CucCloudMessage = CucCloudMessage;
+    this.PciStoragesColdArchiveService = PciStoragesColdArchiveService;
+  }
+
+  $onInit() {
+    this.isLoading = false;
+    this.addExistingUser = 'addExistingUser';
+    this.createNewUser = 'createNewUser';
+    this.users = this.allUserList
+      .filter((user) => user && user.status === 'ok')
+      .map((user) => ({
+        ...user,
+        asCredentials: this.usersCredentials.find(
+          (credential) => credential.userId === user.id,
+        ).asCredentials
+          ? this.$translate.instant(
+              'pci_projects_project_users_add_as_credentials',
+            )
+          : this.$translate.instant(
+              'pci_projects_project_users_add_as_no_credentials',
+            ),
+      }));
+    this.usersWithoutCredentials = this.users
+      .filter(({ s3Credentials }) => !s3Credentials.length)
+      .map((user) => ({
+        ...user,
+        userNameDescriptionKey: user.description
+          ? `${user.username} - ${user.description}`
+          : user.username,
+      }));
+    this.isUserCouldGenerateCredentails =
+      this.usersWithoutCredentials?.length > 0;
+    [this.userModel] = this.usersWithoutCredentials;
+  }
+
+  onChange(action) {
+    this.model = action;
+  }
+
+  onCancel() {
+    this.trackClick(`${TRACKING.ADD_POLICY}::cancel`);
+    return this.cancel();
+  }
+
+  onSubmit(user) {
+    this.trackClick(`${TRACKING.ADD_POLICY}::confirm`);
+    this.isLoading = true;
+    if (!this.isUserCreationModeActive()) {
+      return this.getUserS3Credential(user.id)
+        .then((credentials) => {
+          if (credentials.length === 0)
+            return this.generateUserS3Credential(user);
+
+          return this.goBack(true, user, { ...credentials.pop() }, `-success`);
+        })
+        .finally(() => {
+          this.isLoading = false;
+        });
+    }
+    // action === 'createNewUser'
+    return this.createUser(this.newUserDescription)
+      .then((data) => this.generateUserS3Credential(data))
+      .finally(() => {
+        this.isLoading = false;
+      });
+  }
```

Comment:
**question:** Similar logic is also implemented for `object-storage`. Can we evaluate and create a reusable components?

Reply:
Since this will impact a component outside the scope of this PR, I've created a refactoring task in order to handle this pertinent comment.
I've assigned it to myself and will fix it in a separate PR. 

https://projects.dsi.ovh/browse/MANAGER-10280